service/dap/server_test.go

```diff
@@ -3060,6 +3067,71 @@ func TestBadAttachRequest(t *testing.T) {
 	})
 }
 
+func TestBadInitializeRequest(t *testing.T) {
+	expectFailedToInitialize := func(response *dap.ErrorResponse) {
+		t.Helper()
+		if response.Command != "initialize" {
+			t.Errorf("Command got %q, want \"launch\"", response.Command)
+		}
+		if response.Message != "Failed to initialize" {
+			t.Errorf("Message got %q, want \"Failed to launch\"", response.Message)
+		}
+		if response.Body.Error.Id != 3002 {
+			t.Errorf("Id got %d, want 3002", response.Body.Error.Id)
+		}
+	}
+
+	expectFailedToInitializeWithMessage := func(response *dap.ErrorResponse, errmsg string) {
+		t.Helper()
+		expectFailedToInitialize(response)
+		if response.Body.Error.Format != errmsg {
+			t.Errorf("\ngot  %q\nwant %q", response.Body.Error.Format, errmsg)
+		}
+	}
+
+	runInitializeTest := func(args dap.InitializeRequestArguments, err string) {
+		// Only one initialize request is allowed, so use a new server
+		// for each test.
+		client, teardown := startDapServer(t)
+		defer teardown()
+		client.InitializeRequestWithArgs(args)
+		expectFailedToInitializeWithMessage(client.ExpectErrorResponse(t), err)
```

Comment:
can we inline expectFailedToInitializeWithMessage & expectFailedToInitializeRequest?

Reply:
Done.